test/wrappers/ortools_test.rb

```diff
@@ -5777,4 +5777,39 @@ def test_pause_should_be_last_if_possible
   def test_no_nil_in_corresponding_mission_ids
     assert_empty OptimizerWrapper.config[:services][:ortools].send(:corresponding_mission_ids, ['only_id'], ['non_id'])
   end
+
+  def test_self_selection_should_not_change_vehicle_ids
+    vrp = VRP.independent_skills
+    vrp[:configuration][:preprocessing] = { first_solution_strategy: ['self_selection'] }
+
+    heuristic_counter = 0
+    OptimizerWrapper.stub(
+      :solve,
+      lambda{ |_|
+        heuristic_counter += 1
+        if heuristic_counter == 1
+          # we want to keep best solution, not first one
+          { cost: 27648.0, solvers: ['ortools'], elapsed: 0.94, routes: [{ vehicle_id: 'vehicle_0', activities: [] }, { vehicle_id: 'vehicle_1', activities: [] }, { vehicle_id: 'vehicle_2', activities: [] }] }
+        else
+          { cost: 14.0, solvers: ['ortools'], elapsed: 0.94,
+            routes: [{ vehicle_id: 'vehicle_0', activities: [{ point_id: 'point_0', type: 'depot'}] },
+                     { vehicle_id: 'vehicle_1', activities: [{ point_id: 'point_0', type: 'depot'}, { service_id: 'service_5', point_id: 'point_5', type: 'service'}, { service_id: 'service_3', point_id: 'point_3', type: 'service'}, { service_id: 'service_1', point_id: 'point_1', type: 'service'}, { service_id: 'service_2', point_id: 'point_2', type: 'service' }] },
+                     { vehicle_id: 'vehicle_2', activities: [{ point_id: 'point_0', type: 'depot'}] }] }
+        end
+      }
+    ) do
+      Interpreters::SeveralSolutions.stub(
+        :collect_heuristics,
+        lambda{ |_, _|
+          %w[global_cheapest_arc parallel_cheapest_insertion local_cheapest_insertion savings]
+        }
+      ) do
+        service_vrp = Interpreters::SeveralSolutions.find_best_heuristic({ service: :ortools, vrp: TestHelper.create(vrp) })
+        assert_equal vrp[:vehicles].size, service_vrp[:vrp].vehicles.collect(&:id).uniq.size, 'We expect same number of IDS as initial vehicles in problem'
+        assert_equal vrp[:services].size, service_vrp[:vrp].services.collect(&:id).uniq.size, 'We expect same number of IDS as initial services in problem'
```

Comment:
find_best_heuristic sets the `route`. Can we also verify that the route is as expected -- like we do with vehicles and services? 

At the moment find_best_heurictic doesn't touch the `service_vrp[:vrp].vehicles` nor `service_vrp[:vrp].services` but we can keep them for good measure.

IDS - IDs

Reply:
find_best_heuristic is not SUPPOSED to touch. But the bug we had is that calling this function changed IDs in service_vrp[:vrp].vehicles. So in fact this is the way to test there is no regression about the bug we've had.

Reply:
That is why I want to also test the structure (`:routes`) it touches. It also affected the `routes` and it can happen without touching the `vehicles` or `services` in the future.

Reply:
Yes, I changed the test already